tmt/uuid.py

```diff
@@ -0,0 +1,112 @@
+import re
+from uuid import uuid4
+
+import fmf
+
+import tmt.base
+
+from .utils import log
+
+ID_FMF_ITEM = "id"
+
+
+class IDError(Exception):
+    pass
+
+
+class IDLeafError(IDError):
+    pass
+
+
+def dereference_node(node):
+    """
+    find fmf items inside node
+    """
```

Comment:
I find it quite confusing supporting both `fmf` and `tmt` nodes by the functions. All these methods are basically handling `fmf` metadata only so let's keep the scope to `fmf` nodes only. We should require `fmf.Tree` instances as input. Note that type annotations have been just recently enabled in the repo, perhaps it would be nice to include them here?

Reply:
okay, could be, I've done it, to ensure that `node` is translated to `fmf node` not `tmt node` becuuse in some cases I think I saw, that there is fmf node and in some cases tmt node, so that this unify that it does not matter, what is input, but I undestand your PoV